e2e/tests/desktop/guest-shopper.spec.js

```diff
@@ -28,17 +28,19 @@ test("Guest shopper can checkout items as guest", async ({ page }) => {
     name: /Cotton Turtleneck Sweater/i,
   });
   // selecting swatch
-  const initialImgEl = await productTile.locator("img");
-  const initialSrc = await initialImgEl.getAttribute("src");
+  const productTileImg = productTile.locator("img");
+  const initialSrc = await productTileImg.getAttribute("src");
   await expect(productTile.getByText(/From \$39\.99/i)).toBeVisible();
 
   await productTile.getByLabel(/Black/, { exact: true }).hover();
-  const changedImgEl = await productTile.locator("img");
-  const changeImgSrc = await changedImgEl.getAttribute("src");
+  // Make sure the image src has changed
+  await expect(async () => {
+    const newSrc = await productTileImg.getAttribute("src")
+    expect(newSrc).not.toBe(initialSrc)
+  }).toPass()
```

Comment:
`expect.toPass()` would retry the code block until it passes. See doc at: https://playwright.dev/docs/test-assertions#expectpoll

Reply:
The doc said toPass has timeOut = 0, would it keep running constantly if the newSrc is not found? Should we give it a time out?

Reply:
I was worried about that too, but it turns out that the whole test has its own default timeout already (I believe 30 secs). So it's ok not to specify timeout for `toPass`.